### PR TITLE
[JAX] Use const scale in MHA after softmax

### PR DESCRIPTION
**Pull request by anko-intel**
## Type of Change

feature

## Description

 For tensor quantized inside MultiHeadAttention after softmax const range of values can be assumed. This way calibration or min.max finding for dynamic quantization is not required for this tensor.

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: neural_compressor/jax/quantization/layers_dynamic.py [hunk 1]
```diff
@@ -158,8 +170,36 @@ def call(self, inputs, mask=None):
...
+        """Apply asymmetric quantization using a pre-computed fixed scale.
+
+        Args:
+            inputs (jnp.ndarray): Input tensor.
+            mask (Optional[jnp.ndarray]): Optional mask tensor.
+
+        Returns:
+            jnp.ndarray: Quantized-dequantized tensor.
+        """
+        return self.call_asymmetric(inputs, self._fixed_a_scale, self._fixed_a_zero_point)
```
Comment (bkowalskiINTEL) [inline neural_compressor/jax/quantization/layers_dynamic.py:203]: I think we could skip creating additional methods for fiexd range by adding this small if-else block here.
```suggestion
            if self._fixed_range is None:
                a_scale, a_zero_point = get_q_params(
                batch_min_max, self.activation_dtype, self.compute_dtype, asymmetric=True
            )
            else:
                a_scale = self._fixed_a_scale
                a_zero_point = self._fixed_a_zero_point
            return self.call_asymmetric(inputs, a_scale, a_zero_point)
        a_scale, _ = get_q_params(batch_min_max, self.activation_dtype, self.compute_dtype, asymmetric=False)
        return self.call_symmetric(inputs, a_scale)

```

File: neural_compressor/jax/quantization/layers_dynamic.py [hunk 2]
```diff
@@ -84,46 +86,56 @@ def __init__(self, name, activation_dtype, dtype="float32", asymmetric=False):
...
         self.adequantfun = get_dequantize_fun(dtype=self.compute_dtype, asymmetric=self._is_asymmetric)
+        if self.fixed_range is not None:
+            fixed_min_max = ops.array(self.fixed_range)
+            a_scale, a_zero_point = get_q_params(
+                fixed_min_max, self.activation_dtype, self.compute_dtype, asymmetric=self._is_asymmetric
+            )
+            self._fixed_a_scale = a_scale
+            if self._is_asymmetric:
+                self._fixed_a_zero_point = a_zero_point
+            self.call = self.call_fixed_asymmetric if self._is_asymmetric else self.call_fixed_symmetric
```
Comment (bkowalskiINTEL) [inline neural_compressor/jax/quantization/layers_dynamic.py:111]: We can remove this if we agree on my comment regarding the additional methods

Comment (anko-intel) [reply neural_compressor/jax/quantization/layers_dynamic.py:111]: I rather opt to move all if's to add_variables() and make call function (runtime) as simple as possible. I create separate task for it.

**Review by bkowalskiINTEL — APPROVED** [review]
LGTM
